x-pack/legacy/plugins/reporting/export_types/common/lib/screenshots/index.ts

```diff
@@ -39,108 +28,67 @@ export function screenshotsObservableFactory(
-          ),
-          mergeMap(
-            ({ browser, itemsCount }) =>
-              waitForElementsToBeInDOM(browser, itemsCount, layout, logger),
-            ({ browser }) => browser
-          ),
-          mergeMap(
-            browser => {
-              // Waiting till _after_ elements have rendered before injecting our CSS
-              // allows for them to be displayed properly in many cases
-              return injectCustomCss(browser, layout, logger);
-            },
-            browser => browser
-          ),
-          mergeMap(
-            async browser => {
-              if (layout.positionElements) {
-                // position panel elements for print layout
-                return await layout.positionElements(browser, logger);
-              }
-            },
-            browser => browser
-          ),
-          mergeMap(
-            (browser: HeadlessBrowser) => {
-              return waitForRenderComplete(captureConfig, browser, layout, logger);
-            },
-            browser => browser
-          ),
-          mergeMap(
-            browser => getTimeRange(browser, layout, logger),
-            (browser, timeRange: TimeRange | null) => ({ browser, timeRange })
-          ),
-          mergeMap(
-            ({ browser }) => getElementPositionAndAttributes(browser, layout),
-            ({ browser, timeRange }, elementsPositionAndAttributes: ElementsPositionAndAttribute[]) => {
-              return { browser, timeRange, elementsPositionAndAttributes };
-            } // prettier-ignore
-          ),
-          mergeMap(
-            ({ browser, elementsPositionAndAttributes }) => {
-              return getScreenshots({ browser, elementsPositionAndAttributes, logger });
-            },
-            ({ timeRange }, screenshots) => ({ timeRange, screenshots })
-          )
-        );
+                if (layout.positionElements) {
+                  // position panel elements for print layout
+                  await layout.positionElements(driver, logger);
+                }
 
-        return Rx.race(screenshot$, exit$);
+                await waitForRenderComplete(captureConfig, driver, layout, logger);
+              }),
+              concatMap(() => getTimeRange(driver, layout, logger)),
+              concatMap(
+                async (timeRange): Promise<ScreenshotResults> => {
+                  const elementsPositionAndAttributes = await getElementPositionAndAttributes(
+                    driver,
+                    layout
+                  );
+                  const screenshots = await getScreenshots({
+                    browser: driver,
+                    elementsPositionAndAttributes,
+                    logger,
+                  });
+
+                  return { timeRange, screenshots };
+                }
+              )
+            );
+
+            return Rx.race(screenshot$, exit$);
+          })
+        );
       }),
-      first()
+      toArray()
```

Comment:
Main changes (1) of this PR are above

Reply:
Oops, having the `first` operator is important, so that the observable actually finishes.